redis/reply_test.go

```diff
@@ -225,6 +226,81 @@ func TestSlowLog(t *testing.T) {
 	}
 }
 
+func TestLatency(t *testing.T) {
+	c, err := dial()
+	require.NoError(t, err, "TestLatency failed during dial.")
+	defer c.Close()
+
+	resultStr, err := redis.Strings(c.Do("CONFIG", "GET", "latency-monitor-threshold"))
+	require.NoError(t, err, "TestLatency failed during CONFIG GET latency-monitor-threshold.")
+	// LATENCY commands were added in 2.8.13 so might not be supported.
+	if len(resultStr) == 0 {
+		t.Skip("Latency commands not supported")
+	}
+	latencyMonitorThresholdOldCfg, err := strconv.Atoi(resultStr[1])
+	require.NoError(t, err, "TestLatency failed during strconv.Atoi.")
+	// Enable latency monitoring for events that take 1ms or longer.
+	result, err := c.Do("CONFIG", "SET", "latency-monitor-threshold", "1")
+	require.NoError(t, err)
+	require.Equal(t, "OK", result, "TestLatency failed during CONFIG SET")
+
+	// Sleep for 1ms to register a slow event.
+	_, err = c.Do("DEBUG", "SLEEP", 0.001)
+	require.NoError(t, err, "TestLatency failed during DEBUG SLEEP 0.001")
+	result, err = c.Do("LATENCY", "LATEST")
+	require.NoError(t, err, "TestLatency failed during LATENCY LATEST")
+	latestLatencies, err := redis.Latencies(result, err)
+	require.NoError(t, err, "TestLatency failed during redis.Latencies")
+
+	require.Equal(t, 1, len(latestLatencies))
+	latencyEvent := latestLatencies[0]
+	require.Equal(t, "command", latencyEvent.EventType)
+	require.Equal(t, time.Millisecond, latencyEvent.Latest)
+	require.Equal(t, time.Millisecond, latencyEvent.Max)
```

Comment:
You can combine this into one require which also makes it clearer too:
```suggestion
	expected := []Event{{
		EventType: "command",
		Latest: time.Millisecond,
		Max: time.Millisecond,
	}}
	require.Equal(t, latestLatencies ,expected)
```

Reply:
It would be a great to make this part more readable, however, in the suggestion we are also validating equality of Time which we can't predict in advance.

I've taken a hybrid approach where I am setting expected.Time from the latencyEvent.Time. This way we can collapse 3 requires into 1.